builds/e2e/connectivity.yaml

```diff
@@ -22,14 +48,33 @@ jobs:
     variables:
       edgelet.artifact.name: 'iotedged-ubuntu16.04-amd64'
     steps:
+      - task: Bash@3
+        name: Set_run_flag
+        displayName: 'Set run flag'
+        inputs:
+          targetType: inline
+          script: |
+            output1=$(curl -s -u :$accessToken --request GET "https://dev.azure.com/msazure/one/_apis/build/builds/$buildId?api-version=5.1" | grep -oe '"status":"cancel')
+            output2=$(curl -s -u :$(System.AccessToken) --request GET "https://dev.azure.com/msazure/one/_apis/build/builds/$(Build.BuildId)/Timeline?api-version=5.1" | grep -oe '"result":"canceled"')
+            
+            if [[ ( "$(testrun.config)" == "$(config.to.run)" || "$(config.to.run)" == "0" ) && ( -z "$output1" ) && ( -z "$output2" ) ]]; then
+              echo "##vso[task.setvariable variable=run.steps]1"
+              echo "run.steps=1"
+            else
+              echo "##vso[task.setvariable variable=run.steps]0"
+              echo "run.steps=0"
+            fi
```

Comment:
Just confirming my understanding: This step allows the test to be cancelled before the next matrix entry gets run?

Reply:
All matrix entry will be run and it will skip all steps if build is canceled